code/modules/unit_tests/bloody_footprints.dm

```diff
@@ -0,0 +1,58 @@
+///Tests to make sure bloody footprits work as expected
+///So no stacking, they actually apply, and shoe staining thrown in for free
+/datum/unit_test/bloody_footprints
+
+/datum/unit_test/bloody_footprints/Run()
+	var/mob/living/carbon/human/blood_master = allocate(/mob/living/carbon/human)
+	var/obj/item/clothing/shoes/holds_blood = allocate(/obj/item/clothing/shoes)
+
+	blood_master.equip_to_slot_if_possible(holds_blood, ITEM_SLOT_FEET)
+
+	var/turf/open/move_to = get_step(blood_master, get_dir(blood_master, run_loc_floor_top_right))
+	//We need to not be "on" the same tile as the pool
+	blood_master.forceMove(move_to)
+
+	var/obj/effect/decal/cleanable/blood/pool = allocate(/obj/effect/decal/cleanable/blood)
+
+	//Max out the pools blood, so each step will make things stained enough to matter
+	pool.bloodiness = BLOOD_POOL_MAX
+
+	pool.forceMove(run_loc_floor_bottom_left)
+	blood_master.forceMove(run_loc_floor_bottom_left)
+
+	var/datum/component/bloodysoles/soles = holds_blood.GetComponent(/datum/component/bloodysoles)
+	var/blood_type = pool.blood_state
+
+	TEST_ASSERT(soles.bloody_shoes[blood_type], "Shoes didn't become stained after stepping in a pool of [blood_type]")
```

Comment:
This isn't me testing to see if the shoes have the bloody soles component, this is me testing to see if the shoes actually got bloody after stepping in blood, in order to prevent descriptive failures down the line. 
If you refactor how bloody soles works you'll need to touch this unit test, because the thing is such a web.

Reply:
> this is me testing to see if the shoes actually got bloody after stepping in blood, in order to prevent descriptive failures down the line.

Fair enough.